helper/restore_helper.go

```diff
@@ -225,6 +268,9 @@ func doRestoreAgent() error {
 				readers[contentToRestore], err = getRestoreDataReader(filename, nil, nil)
 				if err != nil {
 					logError(fmt.Sprintf("Oid: %d, Batch %d: Error encountered getting restore data reader: %v", tableOid, batchNum, err))
+					if readers[contentToRestore] != nil {
+						readers[contentToRestore].logPlugin()
+					}
```

Comment:
could we use defer like above?

Reply:
We are getting a new reader for non single data file. In case of a real cluster, we could go over thousands of OIDs, and register thousands defer calls doesn't look right. 

For single data file we reuse reader, so defer is Ok there.

Reply:
Why isn't waitForPlugin always called?

Reply:
We are aborting the execution of the helper right after by returning. In case of zombies, they still will get reaped by `init`. But it can be valid here as well.